### PR TITLE
[CortexXpanse] - EXPANDR 9482 - Update Cortex Xpanse Integration Commands

### PR DESCRIPTION
**Pull request by BigEasyJ**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues

- [EXPANDR-9482](https://jira-dc.paloaltonetworks.com/browse/EXPANDR-9482)
- [EXPANDR-9807](https://jira-dc.paloaltonetworks.com/browse/EXPANDR-9807)

## Description

Updated the following Cortex Xpanse integration commands:
- ***domain***
  - Supports pulling existing domains in XSOAR that are related to Xpanse that were recently updated.
- ***ip***
  - Supports pulling existing IPs in XSOAR that are related to Xpanse that were recently updated.
- ***asm-list-incidents***

Added the ***asm-add-note-to-asset*** Cortex Xpanse integration command

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by BigEasyJ** [comment]
@capanw @johnnywilkes

**Review by johnnywilkes — CHANGES_REQUESTED** [review]
@BigEasyJ , adding notes to:
https://jira-dc.paloaltonetworks.com/browse/EXPANDR-9482
and
https://jira-dc.paloaltonetworks.com/browse/EXPANDR-10221 (I think you meant to reference this 2nd one instead of EXPANDR-9807)

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 1]
```diff
@@ -553,6 +553,22 @@ script:
       required: true
       description: The name of the tags to apply to supplied assets.
     description: Assigns tags to a list of assets.
+  - name: asm-add-note-to-asset
+    description: Adds a note to an asset in Xpanse.
+    arguments:
```
[the remark this answers is not in the document]
Comment (BigEasyJ) [reply Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:558]: I'm open to changing it. I was concerned about customers overwriting by accident, due to them needing to check if comment exist already. Like, User A adds a note in the UI and then User B updates in the API after then overwrites

[the remark this answers is not in the document]
Comment (johnnywilkes) [reply Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:558]: maybe give them the option but default to not override?  Thought is if they might use it for SO info then they might want to override previously assumed SOs that weren't correct

[the remark this answers is not in the document]
Comment (BigEasyJ) [reply Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:558]: Updated in 3b6c0fa

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 2]
```diff
@@ -2714,7 +2900,53 @@ Get a list of all your external websites filtered by authentication type. Maximu
...
+`asm-add-note-to-asset`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| asset_id | Authentication type string on which to search. | Required | 
+| entity_type | Maximum number of assets to return. The default and maximum is 100. | Required | 
+| note_to_add | The custom note to be added to the notes section of the asset in Xpanse | Required |
+| should_append | f you would like to overwrite the current note on the asset, set to 'false'. If you would like to append to the current note, set to 'true' (Note: Default value is 'true'). | Optional |
```
Comment (johnnywilkes) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2922]:
```suggestion
| should_append | If you would like to overwrite the current note on the asset, set to 'false'. If you would like to append to the current note, set to 'true' (Note: Default value is 'true'). | Optional |
```

**Review by johnnywilkes — CHANGES_REQUESTED** [review]

**Comment by BigEasyJ** [comment]
pending approval / review from @johnnywilkes @capanw

**Review by johnnywilkes — APPROVED** [review]
@BigEasyJ , looks good

**Comment by johnnywilkes** [comment]
@ShirleyDenkberg , please review

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 3]
```diff
@@ -553,6 +553,34 @@ script:
       required: true
       description: The name of the tags to apply to supplied assets.
     description: Assigns tags to a list of assets.
+  - name: asm-add-note-to-asset
+    arguments:
+    - name: asset_id
+      description: The asset ID to add a note to.
+      required: true
+    - name: entity_type
+      description: "The type of Xpanse asset. Allowed values are: 'asset' or 'ip_range'."
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:562]:
```suggestion
      description: "The type of Cortex Xpanse asset. 
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 4]
```diff
@@ -553,6 +553,34 @@ script:
...
+      auto: PREDEFINED
+      predefined:
+      - ip_range
+      - asset
+      required: true
+    - name: note_to_add
+      required: true
+      description: The custom note to be added to the notes section of the asset in Xpanse.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:570]:
```suggestion
      description: The custom note to be added to the notes section of the asset in Cortex Xpanse.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 5]
```diff
@@ -553,6 +553,34 @@ script:
...
+    - name: should_append
+      required: false
+      auto: PREDEFINED
+      predefined:
+      - 'True'
+      - 'False'
+      defaultValue: true
+      description: "If you would like to overwrite the current note on the asset, set to 'false'. If you would like to append to the current note, set to 'true' (Note: Default value is 'true')."
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:578]:
```suggestion
      description: Set to 'false' to overwrite the current note on the asset. Set to 'true' to append to the current note. 
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 6]
```diff
@@ -553,6 +553,34 @@ script:
...
+    description: Adds a note to an asset in Xpanse.
+    outputs:
+    - contextPath: ASM.AssetAnnotation.status
+      description: Status of the note being added to the asset in Xpanse.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:582]:
```suggestion
      description: Status of the note being added to the asset in Cortex Xpanse.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 7]
```diff
@@ -570,23 +598,23 @@ script:
   - name: asm-tag-range-remove
     arguments:
     - name: range_id_list
-      description: Comma-separated list of range IDs to remove tags from.
+      description: 'Comma-separated list of range IDs to remove tags from.'
     - name: tags
       description: The name of the tags to remove from supplied IP ranges.
-    description: Removes tags from a list of IP ranges.
+    description: 'Removes tags from a list of IP ranges.'
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:604]: Why are you adding the single quotes here?

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 8]
```diff
@@ -785,17 +813,32 @@ script:
       description: The actual score.
       type: Number
     - contextPath: DBotScore.Reliability
-      description: The reliability of the source providing the intelligence data.
+      description: 'The reliability of the source providing the intelligence data.'
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:816]: Why are you adding the single quotes?

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 9]
```diff
@@ -785,17 +813,32 @@ script:
...
+      description: The existing Xpanse indicator type in XSOAR indicators.
+      type: String
+    - contextPath: ASM.TIM.IP.id  
+      description: The existing indicator ID in XSOAR indicators.
+      type: String
+    - contextPath: ASM.TIM.IP.reliability 
+      description: The existing indicator reliability recently updated in XSOAR indicators.
+      type: String
+    - contextPath: ASM.TIM.IP.score
+      description: The existing indicator score recently updated in XSOAR indicators.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:834]:
```suggestion
      description: The existing Cortex Xpanse IP address recently updated in the Cortex XSOAR indicators.
      type: String
    - contextPath: ASM.TIM.IP.indicator_type
      description: The existing Cortex Xpanse indicator type in the Cortex XSOAR indicators.
      type: String
    - contextPath: ASM.TIM.IP.id  
      description: The existing indicator ID in the Cortex XSOAR indicators.
      type: String
    - contextPath: ASM.TIM.IP.reliability 
      description: The existing indicator reliability recently updated in the Cortex XSOAR indicators.
      type: String
    - contextPath: ASM.TIM.IP.score
      description: The existing indicator score recently updated in the Cortex XSOAR indicators.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml [hunk 10]
```diff
@@ -842,12 +885,27 @@ script:
...
+      description: The existing Xpanse indicator type in XSOAR indicators  .
+      type: String
+    - contextPath: ASM.TIM.Domain.id 
+      description: The existing indicator ID in XSOAR indicators  .
+      type: String
+    - contextPath: ASM.TIM.Domain.reliability 
+      description: The existing indicator reliability recently updated in XSOAR indicators.
+      type: String
+    - contextPath: ASM.TIM.Domain.score 
+      description: The existing indicator score recently updated in XSOAR indicators.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:901]:
```suggestion
      description: The existing Cortex Xpanse domain recently updated in XSOAR indicators.
      type: String
    - contextPath: ASM.TIM.Domain.indicator_type 
      description: The existing Cortex Xpanse indicator type in the Cortex XSOAR indicators  .
      type: String
    - contextPath: ASM.TIM.Domain.id 
      description: The existing indicator ID in the Cortex XSOAR indicators  .
      type: String
    - contextPath: ASM.TIM.Domain.reliability 
      description: The existing indicator reliability recently updated in the Cortex XSOAR indicators.
      type: String
    - contextPath: ASM.TIM.Domain.score 
      description: The existing indicator score recently updated in the Cortex XSOAR indicators.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 11]
```diff
@@ -2147,29 +2278,39 @@ Returns enrichment for a domain.
...
+| DBotScore.Indicator           | String   | The indicator value.                                                       |
+| DBotScore.Type                | String   | The vendor used to calculate the score.                                    |
+| DBotScore.Reliability         | String   | Reliability of the source providing the intelligence data.                 |
+| Domain.Name                   | String   | Name of the domain.                                                        |
+| ASM.TIM.Domain.name           | String   | The existing Xpanse domain recently updated in XSOAR indicators            |
+| ASM.TIM.Domain.indicator_type | String   | The existing Xpanse indicator type in XSOAR indicators                     |
+| ASM.TIM.Domain.id             | String   | The existing indicator ID in XSOAR indicators                              |
+| ASM.TIM.Domain.reliability    | String   | The existing indicator reliability recently updated in XSOAR indicators    |
+| ASM.TIM.Domain.score          | Integer  | The existing indicator score recently updated in XSOAR indicators          |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2304]:
```suggestion
| ASM.TIM.Domain.name           | String   | The existing Cortex Xpanse domain recently updated in the Cortex XSOAR indicators            |
| ASM.TIM.Domain.indicator_type | String   | The existing Cortex Xpanse indicator type in the Cortex XSOAR indicators                     |
| ASM.TIM.Domain.id             | String   | The existing indicator ID in the Cortex XSOAR indicators                              |
| ASM.TIM.Domain.reliability    | String   | The existing indicator reliability recently updated in the Cortex XSOAR indicators    |
| ASM.TIM.Domain.score          | Integer  | The existing indicator score recently updated in the Cortex XSOAR indicators          |
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 12]
```diff
@@ -2025,31 +2103,41 @@ Returns enrichment for an IP address.
...
+| DBotScore.Score           | Integer  | An integer regarding the status of the indicator.                          |
+| DBotScore.Indicator       | String   | The indicator value.                                                       |
+| DBotScore.Type            | String   | The vendor used to calculate the score.                                    |
+| DBotScore.Reliability     | String   | Reliability of the source providing the intelligence data.                 |
+| IP.Address                | String   | IP address.                                                                |
+| ASM.TIM.IP.name           | String   | The existing Xpanse IP address recently updated in XSOAR indicators        |
+| ASM.TIM.IP.indicator_type | String   | The existing Xpanse indicator type in XSOAR indicators                     |
+| ASM.TIM.IP.id             | String   | The existing indicator ID in XSOAR indicators                              |
+| ASM.TIM.IP.reliability    | String   | The existing indicator reliability recently updated in XSOAR indicators    |
+| ASM.TIM.IP.score          | Integer  | The existing indicator score recently updated in XSOAR indicators          |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2130]:
```suggestion
| ASM.TIM.IP.name           | String   | The existing Cortex Xpanse IP address recently updated in the Cortex XSOAR indicators        |
| ASM.TIM.IP.indicator_type | String   | The existing Cortex Xpanse indicator type in the Cortex XSOAR indicators                     |
| ASM.TIM.IP.id             | String   | The existing indicator ID in the Cortex XSOAR indicators                              |
| ASM.TIM.IP.reliability    | String   | The existing indicator reliability recently updated in the Cortex XSOAR indicators    |
| ASM.TIM.IP.score          | Integer  | The existing indicator score recently updated in the Cortex XSOAR indicators          |
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 13]
```diff
@@ -2025,31 +2103,41 @@ Returns enrichment for an IP address.
...
 
 
 #### Command example
-```!ip ip=1.1.1.1```
+
+```!ip ip='1.1.1.1, 1.1.1.2, 8.8.8.8'```
+
 #### Context Example
+
+If the Indicator was **not** updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2139]:
```suggestion
If the indicator was **not** updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 14]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
 }
 ```
 
+If the Indicator is **related to Xpanse** was updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2215]:
```suggestion
If the indicator is **related to Xpanse** was updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 15]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
...
+}
+```
+
 #### Human Readable Output
->### Xpanse IP List
->|asm_ids|asset_explainers|asset_type|domain|domain_details|first_observed|ip|last_observed|name|recent_ips|service_type|tags|
->|---|---|---|---|---|---|---|---|---|---|---|---|
->| 4b1f3765-de40-3a1a-8535-667420408fd9 |  | DOMAIN | *.acme.com | admin: {"city": "", "country": "us", "emailAddress": "", "faxExtension": null, "faxNumber": "", "name": "", "organization": "Acme, Inc.", "phoneExtension": null, "phoneNumber": "", "postalCode": "", "province": "AZ", "registryId": null, "street": ""}| 1679457579382 | 1.1.1.1 | 1697361335282 | *.acme.com | {'id': '218b3cc9-2d26-3a17-aadd-9eac08cc30ec', 'ip': 52529952, 'ipv6': None, 'source': {'name': 'DOMAIN_RESOLUTION'}, 'provider': {'name': 'AWS', 'additionalProviderInfo': None, 'isCdn': False, 'legacyName': 'AWS', 'displayName': 'Amazon Web Services', 'cdn': False}, 'firstObserved': 1692418207732, 'lastObserved': 1697361335282} | HttpServer | BU:Xpanse VanDelay Demo 3 |
+
+If the Indicator was **not** updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2233]:
```suggestion
If the indicator was **not** updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 16]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
...
+
+> ### Xpanse Discovered IP List
+>
+> |asm_ids|asset_explainers|asset_type|domain|domain_details|first_observed|ip|last_observed|name|recent_ips|service_type|tags|
+> |---|---|---|---|---|---|---|---|---|---|---|---|
+> | 4b1f3765-de40-3a1a-8535-667420408fd9 |  | DOMAIN | *.acme.com | admin: {"city": "", "country": "us", "emailAddress": "", "faxExtension": null, "faxNumber": "", "name": "", "organization": "Acme, Inc.", "phoneExtension": null, "phoneNumber": "", "postalCode": "", "province": "AZ", "registryId": null, "street": ""}| 1679457579382 | 1.1.1.1 | 1697361335282 | *.acme.com | {'id': '218b3cc9-2d26-3a17-aadd-9eac08cc30ec', 'ip': 52529952, 'ipv6': None, 'source': {'name': 'DOMAIN_RESOLUTION'}, 'provider': {'name': 'AWS', 'additionalProviderInfo': None, 'isCdn': False, 'legacyName': 'AWS', 'displayName': 'Amazon Web Services', 'cdn': False}, 'firstObserved': 1692418207732, 'lastObserved': 1697361335282} | HttpServer | BU:Xpanse VanDelay Demo 3 |
+
+If the Indicator is **related to Xpanse** was updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2241]:
```suggestion
If the indicator is **related to Xpanse** was updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 17]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
...
+
+> ### Xpanse Discovered IP List (Existing Indicators)
+>
+> This domain list is from existing records found in XSOAR within the last 3 days.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2245]:
```suggestion
> This domain list is from existing records found in Cortex XSOAR within the last 3 days.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 18]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
...
+> If you would additional Xpanse specific information about these please use asm-list-asset-internet-exposure.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2246]:
```suggestion
> If you would additional Cortex Xpanse specific information about these, use asm-list-asset-internet-exposure.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 19]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
...
+>
+> |id|indicator_type|name|reliability|score|
+> |---|---|---|---|---|
+> | abcd1b2abcd1a0b20c7a8bc5d67e8eea | IP | 1.1.1.2 | A+ - 3rd party enrichment | 0 |
+
+If the Indicator was updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2252]:
```suggestion
If the indicator was updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 20]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
...
+
+> ### XSOAR Indicator Discovered IP List (Not Related to Xpanse)
+> 
+> This IP list is from existing records found in XSOAR within the last 3 days.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2256]:
```suggestion
> This IP list is from existing records found in Cortex XSOAR within the last 3 days.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 21]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
...
+> These IPs have not been found to be attributed to Xpanse`.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2257]:
```suggestion
> These IPs have not been found to be attributed to Cortex Xpanse.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 22]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
...
+> |integrations|name|
+> |---|---|
+> | VirusTotal (API v3) | 8.8.8.8 |
+
 
 ### domain
 
 ***
-Returns enrichment for a domain.
+Returns reputation lookup for an doamin found in Xpanse
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2267]:
```suggestion
Returns reputation lookup for an domain found in Cortex Xpanse.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 23]
```diff
@@ -2124,16 +2212,59 @@ Returns enrichment for an IP address.
...
+> This domain list is from existing records found in XSOAR within the last 3 days.
+> If you would additional Xpanse specific information about these please use asm-list-asset-internet-exposure.
+>
+> |id|indicator_type|name|reliability|score|
+> |---|---|---|---|---|
+> | abcd1b2abcd1a0b20c7a8bc5d67e8eea | IP | 1.1.1.2 | A+ - 3rd party enrichment | 0 |
+
+If the Indicator was updated in XSOAR in the last 3 days:
+
+> ### XSOAR Indicator Discovered IP List (Not Related to Xpanse)
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2254]:
```suggestion
> ### XSOAR Indicator Discovered IP List (Not Related to Cortex Xpanse)
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 24]
```diff
@@ -2147,29 +2278,39 @@ Returns enrichment for a domain.
...
+| ASM.TIM.Domain.score          | Integer  | The existing indicator score recently updated in XSOAR indicators          |
 
 #### Command example
-```!domain domain="*.acme.com"```
+
+```!domain domain="*.acme.com, www.example.com, www.fakedomain.com"```
+
 #### Context Example
+
+If the Indicator was **not** updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2312]:
```suggestion
If the indicator was **not** updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 25]
```diff
@@ -2243,11 +2384,53 @@ Returns enrichment for a domain.
 }
 ```
 
+If the Indicator is **related to Xpanse** was updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2387]:
```suggestion
If the indicator is **related to Xpanse** was updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 26]
```diff
@@ -2243,11 +2384,53 @@ Returns enrichment for a domain.
...
+}
+```
+
 #### Human Readable Output
->### Xpanse Domain List
->|asm_ids|asset_explainers|asset_type|domain|domain_details|first_observed|last_observed|name|recent_ips|service_type|tags|
->|---|---|---|---|---|---|---|---|---|---|---|---|
->| 4b1f3765-de40-3a1a-8535-667420408fd9 |  | DOMAIN | *.acme.com | admin: {"city": "", "country": "us", "emailAddress": "", "faxExtension": null, "faxNumber": "", "name": "", "organization": "Acme, Inc.", "phoneExtension": null, "phoneNumber": "", "postalCode": "", "province": "AZ", "registryId": null, "street": ""}| 1679457579382 | 1697361335282 | *.acme.com | {'id': '218b3cc9-2d26-3a17-aadd-9eac08cc30ec', 'ip': 52529952, 'ipv6': None, 'source': {'name': 'DOMAIN_RESOLUTION'}, 'provider': {'name': 'AWS', 'additionalProviderInfo': None, 'isCdn': False, 'legacyName': 'AWS', 'displayName': 'Amazon Web Services', 'cdn': False}, 'firstObserved': 1692418207732, 'lastObserved': 1697361335282} | HttpServer | BU:Xpanse VanDelay Demo 3 |
+
+If the Indicator was **not** updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2405]:
```suggestion
If the indicator was **not** updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 27]
```diff
@@ -2243,11 +2384,53 @@ Returns enrichment for a domain.
...
+
+> ### Xpanse Discovered Domain List
+>
+> |asm_ids|asset_explainers|asset_type|domain|domain_details|first_observed|last_observed|name|recent_ips|service_type|tags|
+> |---|---|---|---|---|---|---|---|---|---|---|---|
+> | 4b1f3765-de40-3a1a-8535-667420408fd9 |  | DOMAIN | *.acme.com | admin: {"city": "", "country": "us", "emailAddress": "", "faxExtension": null, "faxNumber": "", "name": "", "organization": "Acme, Inc.", "phoneExtension": null, "phoneNumber": "", "postalCode": "", "province": "AZ", "registryId": null, "street": ""}| 1679457579382 | 1697361335282 | *.acme.com | {'id': '218b3cc9-2d26-3a17-aadd-9eac08cc30ec', 'ip': 52529952, 'ipv6': None, 'source': {'name': 'DOMAIN_RESOLUTION'}, 'provider': {'name': 'AWS', 'additionalProviderInfo': None, 'isCdn': False, 'legacyName': 'AWS', 'displayName': 'Amazon Web Services', 'cdn': False}, 'firstObserved': 1692418207732, 'lastObserved': 1697361335282} | HttpServer | BU:Xpanse VanDelay Demo 3 |
+
+If the Indicator is **related to Xpanse** was updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2413]:
```suggestion
If the indicator is **related to Xpanse** was updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 28]
```diff
@@ -2243,11 +2384,53 @@ Returns enrichment for a domain.
...
+
+> ### Xpanse Discovered Domain List (Existing Indicators)
+>
+> This domain list is from existing records found in XSOAR within the last 3 days.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2417]:
```suggestion
> This domain list is from existing records found in Cortex XSOAR within the last 3 days.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 29]
```diff
@@ -2243,11 +2384,53 @@ Returns enrichment for a domain.
...
+> If you would additional Xpanse specific information about these please use asm-list-asset-internet-exposure.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2418]:
```suggestion
> If you would like additional Cortex Xpanse specific information about these, use asm-list-asset-internet-exposure.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 30]
```diff
@@ -2243,11 +2384,53 @@ Returns enrichment for a domain.
...
+>
+> |id|indicator_type|name|reliability|score|
+> |---|---|---|---|---|
+> | abcd1b2abcd1a0b20c7a8bc5d67e8eea | Domain | www.example.com | A+ - 3rd party enrichment | 0 |
+
+If the Indicator was updated in XSOAR in the last 3 days:
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2424]:
```suggestion
If the indicator was updated in Cortex XSOAR in the last 3 days:
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 31]
```diff
@@ -2243,11 +2384,53 @@ Returns enrichment for a domain.
...
+
+> ### XSOAR Indicator Discovered Domain List (Not Related to Xpanse)
+> 
+> This domain list is from existing records found in XSOAR within the last 3 days.
+> These domains have not been found to be attributed to Xpanse`.
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2429]:
```suggestion
> This domain list is from existing records found in Cortex XSOAR within the last 3 days.
> These domains have not been found to be attributed to Cortex Xpanse.
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/README.md [hunk 32]
```diff
@@ -2714,7 +2900,53 @@ Get a list of all your external websites filtered by authentication type. Maximu
...
+`asm-add-note-to-asset`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| asset_id | Authentication type string on which to search. | Required | 
+| entity_type | Maximum number of assets to return. The default and maximum is 100. | Required | 
+| note_to_add | The custom note to be added to the notes section of the asset in Xpanse | Required |
+| should_append | If you would like to overwrite the current note on the asset, set to 'false'. If you would like to append to the current note, set to 'true' (Note: Default value is 'true'). | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/Integrations/CortexXpanse/README.md:2922]:
```suggestion
| note_to_add | The custom note to be added to the notes section of the asset in Cortex Xpanse | Required |
| should_append | Set to 'false' to overwrite the current note on the asset. Set to 'true' to append to the current note. Default is 'true'. | Optional |
```

File: Packs/CortexXpanse/ReleaseNotes/1_2_1.md [hunk 33]
```diff
@@ -0,0 +1,13 @@
...
+##### Cortex Xpanse
+
+Updated the following Cortex Xpanse integration commands:
+- ***domain***
+  - Supports pulling existing domains in XSOAR that are related to Xpanse that were recently updated.
+- ***ip***
+  - Supports pulling existing IPs in XSOAR that are related to Xpanse that were recently updated.
+- ***asm-list-incidents***
+
+Added the ***asm-add-note-to-asset*** Cortex Xpanse integration command
```
Comment (ShirleyDenkberg) [inline Packs/CortexXpanse/ReleaseNotes/1_2_1.md:13]:
```suggestion
- Updated the following Cortex Xpanse integration commands:
   - ***domain***- Supports pulling existing domains in Cortex XSOAR that are related to Cortex Xpanse that were recently updated.
   - ***ip***- Supports pulling existing IPs in Cortex XSOAR that are related to Cortex Xpanse that were recently updated.
   - ***asm-list-incidents***- Fetches ASM incidents that match provided filters. Incidents are an aggregation of related alerts. 

- Added the ***asm-add-note-to-asset*** Cortex Xpanse integration command
```

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml (hunk 3 above)
Comment (BigEasyJ) [reply Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:562]: Only those 2 types are allowed. I don't see why we would remove them from the description the user sees.

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml (hunk 8 above)
Comment (BigEasyJ) [reply Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:816]: Autoformat did

File: Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml (hunk 7 above)
Comment (BigEasyJ) [reply Packs/CortexXpanse/Integrations/CortexXpanse/CortexXpanse.yml:604]: Autoformat did

**Comment by BigEasyJ** [comment]
@thefrieddan1 @MLainer1 ready for merging please and thank you

**Review by thefrieddan1 — APPROVED** [review]

**Pull request merged**